### PR TITLE
templates: Add a release check list template

### PR DESCRIPTION
**Pull request by fidencio**
Let's have a release check list template that we can use to track the release steps needed for each new release.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release-check-list.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+name: New Release Check List
+about: v<TARGET_RELEASE>
+title: "v<TARGET_RELEASE Check List"
+---
+
+## v<TARGET_RELEASE>
+
+* [ ] - Get the latest ocicrypt-rs used as part of image-rs
```
Comment (stevenhorsman) [inline .github/ISSUE_TEMPLATE/release-check-list.md:9]: Sorry, just to clarify - by `Get the latest ocicrypt-rs used as part of image-rs` do you mean identify the commit of ocicrypt-rs that is used by the version of image-rs that the kata-agent is pulling in, or something else - like bump the version of ocicrypt-rs that image-rs is using to the latest ocicrypt-rs sha (and then test it)?

File: .github/ISSUE_TEMPLATE/release-check-list.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+* [ ] - Get the image-rs, using the latest ocicrypt-rs, as part of attestastion-agent
```
Comment (stevenhorsman) [inline .github/ISSUE_TEMPLATE/release-check-list.md:10]: Maybe I'm reading this/understanding it incorrrectly, but I think that the attestation-agent is part of image-rs, not the other way around? This cascades to the next for items as well.

File: .github/ISSUE_TEMPLATE/release-check-list.md (hunk 1 above)
Comment (stevenhorsman) [reply .github/ISSUE_TEMPLATE/release-check-list.md:9]: Is it also worth for these mentioned where/how they are pulled in, e.g. in Cargo.toml in this case to point any less experienced people in the right direction?

File: .github/ISSUE_TEMPLATE/release-check-list.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+- [ ] - Get the latest attestation-agent, using the latest image-rs, as part of the Enclave CC
+- [ ] - Get the latest attestation-agent, using the latest image-rs, as part of the Kata Containers
```
Comment (stevenhorsman) [inline .github/ISSUE_TEMPLATE/release-check-list.md:12]: So we pull in image-rs and the attestation-agent independently in Kata Containers:
- image-rs is pulled in via the kata-agent's Cargo.toml (twice, once for non-s390x and the other for s390x)
- attestation-agent via the versions.yaml file
I think we want to reflect that in this bullet?

File: .github/ISSUE_TEMPLATE/release-check-list.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+- [ ] - Get the latest td-shim as part of the Kata Containers
+- [ ] - Get the latest Enclave CC payload tested as part of the Operator
+- [ ] - Get the latest Kata Containers payload tested as part of the Operator
+- [ ] - Cut an ocicrypt-rs v<TARGET_RELEASE> release
+- [ ] - Cut an image-rs v<TARGET_RELEASE> release, using ocicrypt-rs v<TARGET_RELEASE> release
+- [ ] - Cut an attestation-agent v<TARGET_RELEASE> release, using the image-rs v<TARGET_RELEASE> release
+- [ ] - Cut a td-shim v<TARGET_RELEASE> release
```
Comment (stevenhorsman) [inline .github/ISSUE_TEMPLATE/release-check-list.md:19]: Do we need an extra step here to cover updating kata-containers to use the tagged release of the attestation-agent and image-rs once they've been cut, but before we've tagged the Kata containers

**Review by stevenhorsman — COMMENTED** [review]
I have a few questions on some specific items and wonder if we need to provide more details and help for a less experienced person to follow, but thanks for the initial list - it's super helpful and I don't mean to poke holes!

**Comment by fidencio** [comment]
@stevenhorsman, @mythi, please, give it a second look as I updated pretty much the whole document.

File: .github/ISSUE_TEMPLATE/release-check-list.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+- [ ] - 6. Update Kata Containers to use the latest td-shim
+        * https://github.com/kata-containers/kata-containers/blob/CCv0/versions.yaml
+          * Change the version
+- [ ] - 7. Check if there are new changes in the pre install payload script
+        * https://github.com/confidential-containers/operator/tree/main/install/pre-install-payload
+          * The last commit there must match what's in the following files as preInstall / postUninstall image
+            * Enclave CC: https://github.com/confidential-containers/operator/blob/main/config/samples/enclave-cc/base/ccruntime-enclave-cc.yaml
+            * Kata Containers:
+              * Note that for Kata Containers, we're looking for the newTag, below the quay.io/confidential-containers/container-engine-for-cc-payload image
+              * s390x: https://github.com/confidential-containers/operator/blob/main/config/samples/ccruntime/s390x/kustomizat
```
Comment (stevenhorsman) [inline .github/ISSUE_TEMPLATE/release-check-list.md:41]:
```suggestion
              * s390x: https://github.com/confidential-containers/operator/blob/main/config/samples/ccruntime/s390x/kustomization.yaml
```

File: .github/ISSUE_TEMPLATE/release-check-list.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+          * Note that we need the quay.io/confidential-containers/runtime-payload-ci registry and kata-containers-latest tag
+- [ ] - 9. Cut an ocicrypt-rs v<TARGET_RELEASE> release, if changes happened in the project
+- [ ] - 10. Cut an attestation-agent v<TARGET_RELEASE>, if changes happened in the project
+- [ ] - 11. Cut an image-rs v<TARGET_RELEASE> release, using the latest release of:
+        * ocicrypt-rs (redo step 1, but now using v<TARGET_RELEASE>)
+        * attestation-agent (redo step 2, but now using v<TARGET_RELEASE>)
+- [ ] - 12. Cut a td-shim v<TARGET_RELEASE> release, if changes happened in the project
+- [ ] - 13. Update Enclave CC to use the released version of image-rs
+        * redo step 3, but now using v<TARGET_RELEASE>
+- [ ] - 14. Update the operator to use the images generated from the latest commit of both Kata Containers and Enclave CC
```
Comment (stevenhorsman) [inline .github/ISSUE_TEMPLATE/release-check-list.md:60]: Do we need to also redo step 4 and 5 using the released versions of image-rs and the attestation-agent?

File: .github/ISSUE_TEMPLATE/release-check-list.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        * redo step 8, but now targetting the latest payload image generated for Kata Containers and Enclave CC
+- [ ] - 15. Make sure all the operator tests are passing
+- [ ] - 16. Cut an Enclave CC release
+- [ ] - 17. Add a new Kata Containers tag
+- [ ] - 18. Update the operator to use the release tags coming from Enclave CC and Kata Containers
+        * redo step 8, but now targetting thje latest release of the payload image generated for Kata Containers eand Enclave CC
+- [ ] - 19. Update the Operator version
+        * https://github.com/confidential-containers/operator/blob/main/config/release/kustomization.yaml#L7
+- [ ] - 20. Cut an operator release
```
Comment (stevenhorsman) [inline .github/ISSUE_TEMPLATE/release-check-list.md:69]: This isn't something I've been involved in, so might be incorrect, but I think Jens normally updates Operatorhub, so maybe we should add a step to ask him to do that?

**Review by stevenhorsman — COMMENTED** [review]
It's looking much clearer. Thanks a lot for the changes. Just a few extra suggestions...

File: .github/ISSUE_TEMPLATE/release-check-list.md (hunk 7 above)
Comment (fidencio) [reply .github/ISSUE_TEMPLATE/release-check-list.md]: Added!

File: .github/ISSUE_TEMPLATE/release-check-list.md (hunk 6 above)
Comment (fidencio) [reply .github/ISSUE_TEMPLATE/release-check-list.md]: We do, fixed!

File: .github/ISSUE_TEMPLATE/release-check-list.md (hunk 5 above)
Comment (fidencio) [reply .github/ISSUE_TEMPLATE/release-check-list.md]: Done, thanks!

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks very much for getting this info doc'd!

File: .github/ISSUE_TEMPLATE/release-check-list.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
+# v<TARGET_RELEASE>
+
+## Code freeze
+
+* [ ] - 1. Update image-rs to use the latest commit from ocicrypt-rs
```
Comment (sameo) [inline .github/ISSUE_TEMPLATE/release-check-list.md:5]: I think you want: 
```suggestion
- [ ] - 1. Update image-rs to use the latest commit from ocicrypt-rs
```

**Review by sameo — APPROVED** [review]
Just a minor nit. Thanks a lot @fidencio

File: .github/ISSUE_TEMPLATE/release-check-list.md (hunk 8 above)
Comment (fidencio) [reply .github/ISSUE_TEMPLATE/release-check-list.md]: Fixed, thanks!

**Pull request merged**